Mage/src/main/java/mage/abilities/ActivatedAbility.java

```diff
@@ -7,41 +7,51 @@
 import mage.constants.TimingRule;
 import mage.game.Game;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 import java.util.UUID;
 
 /**
- * @author BetaSteward_at_googlemail.com
+ * @author BetaSteward_at_googlemail.com, Susucr
  */
 public interface ActivatedAbility extends Ability {
 
     final class ActivationStatus {
 
-        private final boolean canActivate;
-        private final ApprovingObject approvingObject;
+        private final Set<ApprovingObject> approvingObjects;
 
-        public ActivationStatus(boolean canActivate, ApprovingObject approvingObject) {
-            this.canActivate = canActivate;
-            this.approvingObject = approvingObject;
+        // If true, the Activation Status will not check if there is an approvingObject.
+        private final boolean forcedCanActivate;
+
+        public ActivationStatus(ApprovingObject approvingObject) {
+            this(new HashSet<>(Arrays.asList(approvingObject)));
```

Comment:
is this the best pattern to initialize for a single object? seems like an extra layer in there somewhere... maybe doesn't really matter though

Reply:
If you have a better way to inline make a Set of one element, I'll take it. being called in a this does make it harder.

Reply:
Well, do you need to call a `this()`? I know it's generally good style for consistency, but it's not a complicated constructor. How about: 
```java
public ActivationStatus(ApprovingObject approvingObject) {
    this.forcedCanActivate = false;
    this.approvingObjects = Collections.singleton(approvingObject);
}
```

This is assuming you don't ever need to mutate the set (which I think is true but haven't confirmed).

Reply:
There should be no mutation after the ActivationStatus is created.